src/docmanager/index.ts

```diff
@@ -399,7 +442,10 @@ class DocumentManager {
   }
 
   private _data: { [key: string]: Private.IDocumentData } = Object.create(null);
-  private _fileTypes: { [key: string ]: string[] } = Object.create(null);
+  private _modelFactories: { [key: string]: IModelFactory } = Object.create(null);
```

Comment:
oh, and typescript doesn't compile down `for (let i of object)` if the object is not an array or string. But `for (let i in object)` still works, especially if the object has a null prototype.